### PR TITLE
New Adapter: RevX

### PR DESCRIPTION
**Pull request by nikithacode**

### CODE REVIEW

**Comment by nikithacode** [comment]
Hi @bsardo could you please review this

**Comment by bsardo** [comment]
@pm-jaydeep-mohite can you please review?

File: adapters/revx/revx.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	return bidResponse, nil
+}
+
+func getMediaTypeForImp(bid openrtb2.Bid) (openrtb_ext.BidType, error) {
+	// Check if MType is 0 (invalid or missing media type)
+	if bid.MType == 0 {
+		return "", &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unsupported mtype %d for bid %s", bid.MType, bid.ID),
+		}
+	}
```
Comment (pm-jaydeep-mohite) [inline adapters/revx/revx.go]: Do we need this extra check as default: case can handle this ?

File: static/bidder-info/revx.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# static/bidder-info/revx.yaml
+
+endpoint:  "prebid-use.atomex.net?ag=%s"
+
+maintainer:
+  email: prebidsupport@revx.io
```
Comment (pm-jaydeep-mohite) [inline static/bidder-info/revx.yaml:3]: Sent verification email, please respond to it.

File: adapters/revx/revx.go (hunk 1 above)
Comment (nikithacode) [reply adapters/revx/revx.go]: Hi, you're right — the default case covers it. We'll remove the extra check and update the review. Thanks!

File: static/bidder-info/revx.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# static/bidder-info/revx.yaml
+
+endpoint:  "prebid-use.atomex.net?ag=%s"
```
Comment (ccorbo) [inline static/bidder-info/revx.yaml]: issue: this is an invalid url. You can set the url as "prebid-use.atomex.net" and add query params as need in your adapter code

File: adapters/revx/revx_test.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+		t.Logf("RevX Builder created successfully: %+v", bidder)
+	}
+
+	adapterstest.RunJSONBidderTest(t, "revxtest", bidder) // Call it directly
+
+	t.Logf("--- Finished TestJsonSamples for revxtest/exemplary/simple-app-banner.json ---")
+
+	adapterstest.RunJSONBidderTest(t, "revxtest", bidder)
+
```
Comment (ccorbo) [inline adapters/revx/revx_test.go]: Why is this being ran twice? I think it can be removed

File: adapters/revx/revx.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+	var revxExt openrtb_ext.ExtImpRevX
+	if err := jsonutil.Unmarshal(bidderExt.Bidder, &revxExt); err != nil {
+		errors = append(errors, &errortypes.BadInput{Message: "bad revx bidder ext"})
+
+	}
+
+	// Check if publisher name is present
+	if len(revxExt.PubName) == 0 {
```
Comment (ccorbo) [inline adapters/revx/revx.go]: since you have PubName as a required param, it is guranteed to be present at this point. No need for this check

File: adapters/revx/revx.go [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+	headers.Add("Accept", "application/json")
+
+	// Marshal the OpenRTB bid request into JSON
+	reqJson, err := jsonutil.Marshal(request)
+	if err != nil {
+		return nil, []error{&errortypes.BadInput{Message: fmt.Sprintf("Failed to marshal request: %s", err)}}
+	}
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
```
Comment (ccorbo) [inline adapters/revx/revx.go]: nit: use http.MethodPost

File: adapters/revx/revx.go [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+		endPoint: config.Endpoint, // Default endpoint
+	}, nil
+}
+
+// MakeRequests handles the OpenRTB bid request and returns адаптер.RequestData
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var requests []*adapters.RequestData
+	var errors []error
+
+	if len(request.Imp) == 0 {
```
Comment (ccorbo) [inline adapters/revx/revx.go]: Request is guranteed to have imps here, no need for this check
https://github.com/prebid/prebid-server/blob/master/exchange/bidder.go#L158

File: adapters/revx/revx.go [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+// MakeBids handles the OpenRTB bid response.
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if externalRequest == nil {
+		return nil, nil
+	}
+
+	// Treat 204 and 400 as no-bid without error
+	if response.StatusCode == http.StatusNoContent || response.StatusCode == http.StatusBadRequest {
```
Comment (ccorbo) [inline adapters/revx/revx.go]: Nit: please use util functions for status code checks:
https://github.com/prebid/prebid-server/blob/master/adapters/response.go#L10

File: static/bidder-info/revx.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+maintainer:
+  email: prebidsupport@revx.io
+
+gvlVendorID: 375
```
Comment (pm-jaydeep-mohite) [inline static/bidder-info/revx.yaml:4]: Verified

Comment (nikithacode) [reply static/bidder-info/revx.yaml:4]: Hi Could you send the email again

File: adapters/revx/revx.go (hunk 8 above)
Comment (nikithacode) [reply adapters/revx/revx.go]: updated

File: adapters/revx/revx.go (hunk 7 above)
Comment (nikithacode) [reply adapters/revx/revx.go]: by removing this receiving code coverage build issue

File: adapters/revx/revx.go (hunk 6 above)
Comment (nikithacode) [reply adapters/revx/revx.go]: updated

File: adapters/revx/revx.go (hunk 5 above)
Comment (nikithacode) [reply adapters/revx/revx.go]: by removing this receiving code coverage build issue

File: adapters/revx/revx_test.go (hunk 4 above)
Comment (nikithacode) [reply adapters/revx/revx_test.go]: updated removed the extra lines

File: static/bidder-info/revx.yaml (hunk 3 above)
Comment (nikithacode) [reply static/bidder-info/revx.yaml]: updated the endpoint

File: static/bidder-info/revx.yaml (hunk 2 above)
Comment (nikithacode) [reply static/bidder-info/revx.yaml:3]: could you pls send the email again for verification

File: adapters/revx/revx.go (hunk 1 above)
Comment (nikithacode) [reply adapters/revx/revx.go]: updated the review

File: adapters/revx/revx.go (hunk 5 above)
Comment (ccorbo) [reply adapters/revx/revx.go]: you're code coverage doesnt even include this block, just that it hits this statement. You should be able to make up the code coverage in other places. Same with the other block

File: adapters/revx/revx.go (hunk 7 above)
Comment (ccorbo) [reply adapters/revx/revx.go]: still is not needed - just make up the code coverage elsewhere

File: adapters/revx/revx.go [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}
+	}
+
+	return nil
+}
```
Comment (ccorbo) [inline adapters/revx/revx.go]: Why did you rewrite this function? Please use the function from adatpers/response.go

File: static/bidder-info/revx.yaml (hunk 9 above)
Comment (pm-jaydeep-mohite) [reply static/bidder-info/revx.yaml:4]: Sent it again

File: static/bidder-info/revx.yaml (hunk 2 above)
Comment (pm-jaydeep-mohite) [reply static/bidder-info/revx.yaml:3]: Sent it again

Comment (pm-jaydeep-mohite) [reply static/bidder-info/revx.yaml:3]: Verified

File: static/bidder-info/revx.yaml (hunk 9 above)
Comment (pm-jaydeep-mohite) [reply static/bidder-info/revx.yaml:4]: Verified email id

File: adapters/revx/revx.go [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if externalRequest == nil {
+		return nil, nil
+	}
+
+	// Check HTTP status before parsing response body
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		// Treat 204 and 400 as no-bid without logging error
+		if response.StatusCode == http.StatusNoContent || response.StatusCode == http.StatusBadRequest {
+			return nil, nil
```
Comment (pm-jaydeep-mohite) [inline adapters/revx/revx.go]: Is it ok if adaptor level metrics for StatusBadRequest does not get logged as we are returning err=nil  ?

File: adapters/revx/revx.go [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+	var serverBidResponse openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(response.Body, &serverBidResponse); err != nil {
+		return nil, []error{err}
+	}
+
+	var typedBids []*adapters.TypedBid
+	for _, sb := range serverBidResponse.SeatBid {
+		for i := range sb.Bid {
+			mediaType, err := getMediaTypeForImp(sb.Bid[i])
+			if err != nil {
```
Comment (pm-jaydeep-mohite) [inline adapters/revx/revx.go:80]: Is it Ok to discard all the bids even in case of single bid is invalid ?

**Review by pm-jaydeep-mohite — COMMENTED** [review]
Added few comments

**Comment by nikithacode** [comment]
Hi @pm-jaydeep-mohite could you please review the updated changes

File: adapters/revx/revx.go [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{Message: "No valid impressions for grid"}}
+	}
+
+	imp := request.Imp[0]
+
+	// Parse imp.ext
+	var bidderExt adapters.ExtImpBidder
+	if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		errors = append(errors, &errortypes.BadInput{Message: fmt.Sprintf("invalid imp.ext: %s", err)})
```
Comment (ccorbo) [inline adapters/revx/revx.go]: please add supplemental test where unmarshal results in error

**Comment by nikithacode** [comment]
hi @pm-jaydeep-mohite @bsardo could you please review the updated code changes

File: adapters/revx/revx.go [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+
+	// For all other status codes
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
+	}
+
+	// Proceed with normal parsing
+	var serverBidResponse openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(response.Body, &serverBidResponse); err != nil {
+		return nil, []error{err}
```
Comment (ccorbo) [inline adapters/revx/revx.go:72]: please add supplemental test to  cover this unmarshal error

File: adapters/revx/revx.go [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+	if response.StatusCode == http.StatusBadRequest {
+		// ✅ Return error to log 400 metrics
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Received 400 Bad Request from endpoint: %s", a.endPoint),
+		}}
+	}
+
+	// For all other status codes
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
```
Comment (ccorbo) [inline adapters/revx/revx.go:66]: please add supplemental test to  cover this error path

File: adapters/revx/revx.go [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}
+
+	return []*adapters.RequestData{requestData}, errors
+}
+
+// MakeBids handles the OpenRTB bid response.
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if externalRequest == nil {
+		return nil, nil
```
Comment (ccorbo) [inline adapters/revx/revx.go:56]: please add supplemental test to cover this case where externalRequest == nil

File: adapters/revx/revx.go (hunk 13 above)
Comment (asifrahiman) [reply adapters/revx/revx.go]: added file imp_ext_unmarshal.json

File: adapters/revx/revx.go (hunk 14 above)
Comment (asifrahiman) [reply adapters/revx/revx.go:72]: added file response_unmarshal_error.json

File: adapters/revx/revx.go (hunk 15 above)
Comment (asifrahiman) [reply adapters/revx/revx.go:66]: added file 500Response.json

File: adapters/revx/revx.go (hunk 16 above)
Comment (asifrahiman) [reply adapters/revx/revx.go:56]: added file external_request_null.json

File: static/bidder-info/revx.yaml [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+
+maintainer:
+  email: prebidsupport@revx.io
+
+gvlVendorID: 375  
+gdpr:
+  enabled: true
+  consent_required: true
+  default_value: 0  # Default to 'no consent' if not provided
+  vendor_id: 100  # Vendor ID for your GDPR configuration
```
Comment (bsardo) [inline static/bidder-info/revx.yaml]: Please remove these fields as they are not valid. Fields like `gdpr.enabled`, `gdpr.default_value` and `gdpr.host_vendor_id` exist on the host configuration which you would specify in a `pbs.json` or `pbs.yaml` config file or via environment variables.

File: static/bidder-info/revx.yaml [hunk 18]
```diff
@@ -0,0 +1,24 @@
+# static/bidder-info/revx.yaml
+
+endpoint:  "https://prebid-use.atomex.net"
```
Comment (bsardo) [inline static/bidder-info/revx.yaml]: Endpoint is not reachable as this is timing out:
```
curl -i --location --request POST https://prebid-use.atomex.net
HTTP/1.1 504 Gateway Time-out
Date: Mon, 22 Sep 2025 18:31:56 GMT
Content-Type: text/html
Content-Length: 173
Connection: keep-alive

<html>
<head><title>504 Gateway Time-out</title></head>
<body>
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>openresty/1.17.8.2</center>
</body>
</html>
```

File: static/bidder-info/revx.yaml [hunk 19]
```diff
@@ -0,0 +1,24 @@
+# static/bidder-info/revx.yaml
+
```
Comment (bsardo) [inline static/bidder-info/revx.yaml]: Delete these lines

File: adapters/revx/revx.go [hunk 20]
```diff
@@ -0,0 +1,150 @@
...
+		return nil, errors
+	}
+
+	// Validate impression array
+	if len(request.Imp) == 0 {
+		errors = append(errors, &errortypes.BadInput{
+			Message: "no impression objects provided in request",
+		})
+		return nil, errors
+	}
```
Comment (bsardo) [inline adapters/revx/revx.go]: As mentioned [here](https://github.com/prebid/prebid-server/pull/4335/files#r2172062411), you should remove this imp length check. Your adapter will only ever be called if there is at least one valid imp.

File: adapters/revx/revx.go [hunk 21]
```diff
@@ -0,0 +1,150 @@
...
+	if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		errors = append(errors, &errortypes.BadInput{Message: fmt.Sprintf("invalid imp.ext: %s", err)})
+		return nil, errors
+	}
+
+	var revxExt openrtb_ext.ExtImpRevX
+	if err := jsonutil.Unmarshal(bidderExt.Bidder, &revxExt); err != nil {
+		errors = append(errors, &errortypes.BadInput{Message: "Publisher name missing"})
+		return nil, errors
+	}
```
Comment (bsardo) [inline adapters/revx/revx.go]: It looks like you're just unpacking your bidder param so you can validate it; you aren't using it anywhere in your adapter. This is unnecessary as your adapter will only ever be called if your bidder params are valid in accordance with the rules you have defined in your bidder-info YAML file `static/bidder-info/revx.yaml`. Please remove all of this logic.

Comment (asifrahiman) [reply adapters/revx/revx.go]: @bsardo This is done

File: adapters/revx/revx.go (hunk 20 above)
Comment (asifrahiman) [reply adapters/revx/revx.go]: @bsardo This is done

File: static/bidder-info/revx.yaml (hunk 19 above)
Comment (asifrahiman) [reply static/bidder-info/revx.yaml]: @bsardo This is done

File: static/bidder-info/revx.yaml (hunk 17 above)
Comment (asifrahiman) [reply static/bidder-info/revx.yaml]: @bsardo This is done

File: static/bidder-info/revx.yaml (hunk 18 above)
Comment (asifrahiman) [reply static/bidder-info/revx.yaml]: @bsardo This endpoint is up and running.
```
ping prebid-use.atomex.net
PING gcpadapter-use.atomex.net (35.221.20.134): 56 data bytes
64 bytes from 35.221.20.134: icmp_seq=0 ttl=59 time=260.587 ms
64 bytes from 35.221.20.134: icmp_seq=1 ttl=59 time=259.648 ms
64 bytes from 35.221.20.134: icmp_seq=2 ttl=59 time=271.844 ms
64 bytes from 35.221.20.134: icmp_seq=3 ttl=59 time=260.166 ms
```

File: adapters/revx/revx.go (hunk 16 above)
Comment (ccorbo) [reply adapters/revx/revx.go:56]: did you missing adding this file? I don't see it committed

**Comment by asifrahiman** [comment]
@bsardo @ccorbo @pm-jaydeep-mohite
Gentle reminder to please revisit the PR
Thank you for your time and support!

File: adapters/revx/revx.go (hunk 16 above)
Comment (ccorbo) [reply adapters/revx/revx.go:56]: this still isn't covered, did you miss pushing this file?

File: adapters/revx/revxtest/supplemental/invalid-mtype copy.json [hunk 22]
```diff
@@ -0,0 +1,94 @@
+{
```
Comment (ccorbo) [inline adapters/revx/revxtest/supplemental/invalid-mtype copy.json]: issue: theres no difference between this file and invalid-mtype.json, can you please remove this

File: adapters/revx/revx.go (hunk 16 above)
Comment (asifrahiman) [reply adapters/revx/revx.go:56]: @ccorbo 
Apologies, 
I have added adapters/revx/revx_external_null_test.go file to test the same.

**Comment by asifrahiman** [comment]
@ccorbo 
When running ./validate.sh we are getting a new error
`ok  	github.com/prebid/prebid-server/v3/errortypes	(cached)
2025/10/29 23:52:34 http: superfluous response.WriteHeader call from github.com/prebid/prebid-server/v3/exchange.TestExtraBidWithMultiCurrencies.func1 (bidder_test.go:2991)
--- FAIL: TestNewExchange (0.04s)
    exchange_test.go:91: NewExchange produced an Exchange without bidder revx
E1029 23:52:34.488998   33819 exchange.go:909] OpenRTB auction recovered panic from Bidder appnexus: panic!. Account id: test1, All Bidders: bidder1,bidder2, Stack trace is: goroutine 811 [running]:
runtime/debug.Stack()
	/opt/homebrew/Cellar/go/1.24.3/libexec/src/runtime/debug/stack.go:26 +0x64
github.com/prebid/prebid-server/v3/exchange.TestPanicRecovery.(*exchange).recoverSafely.func2.1()
	/Users/asifabdulrahiman/Documents/releases/2025/May/PreBidServer/prebid-server/exchange/exchange.go:911 +0x170
panic({0x102d61720?, 0x102f4e4c0?})
	/opt/homebrew/Cellar/go/1.24.3/libexec/src/runtime/panic.go:792 +0x124
github.com/prebid/prebid-server/v3/exchange.TestPanicRecovery.func1({0x1400081c000, {0x102abbe5c, 0x7}, {0x102abd685, 0x8}, {{0x102ab8234, 0x3}, {0x102ac1d38, 0xc}, {0x102abd685, ...}, ...}, ...}, ...)
	/Users/asifabdulrahiman/Documents/releases/2025/May/PreBidServer/prebid-server/exchange/exchange_test.go:1943 +0x2c
github.com/prebid/prebid-server/v3/exchange.TestPanicRecovery.(*exchange).recoverSafely.func2({0x1400081c000, {0x102abbe5c, 0x7}, {0x102abd685, 0x8}, {{0x102ab8234, 0x3}, {0x102ac1d38, 0xc}, {0x102abd685, ...}, ...}, ...}, ...)
	/Users/asifabdulrahiman/Documents/releases/2025/May/PreBidServer/prebid-server/exchange/exchange.go:919 +0xbc
github.com/prebid/prebid-server/v3/exchange.TestPanicRecovery(0x140002e36c0)
	/Users/asifabdulrahiman/Documents/releases/2025/May/PreBidServer/prebid-server/exchange/exchange_test.go:1974 +0x488
testing.tRunner(0x140002e36c0, 0x102f4a070)
	/opt/homebrew/Cellar/go/1.24.3/libexec/src/testing/testing.go:1792 +0xe4
created by testing.(*T).Run in goroutine 1
	/opt/homebrew/Cellar/go/1.24.3/libexec/src/testing/testing.go:1851 +0x374
E1029 23:52:35.601688   33819 exchange.go:774] Exchange: bidlables for pubmatic (pubmatic) missing adapter string
E1029 23:52:35.601984   33819 exchange.go:774] Exchange: bidlables for pubmatic (pubmatic) missing adapter string
E1029 23:52:35.602102   33819 exchange.go:774] Exchange: bidlables for pubmatic (pubmatic) missing adapter string
E1029 23:52:35.602207   33819 exchange.go:774] Exchange: bidlables for pubmatic (pubmatic) missing adapter string
FAIL
FAIL	github.com/prebid/prebid-server/v3/exchange	6.942s
`
Could you let me know how to fix the same?

File: adapters/revx/revxtest/supplemental/invalid-mtype copy.json (hunk 22 above)
Comment (asifrahiman) [reply adapters/revx/revxtest/supplemental/invalid-mtype copy.json]: @ccorbo 
This file is removed.

**Comment by ccorbo** [comment]
@asifrahiman
You need to add a revx.yaml file under static/bidder-info. I see it was present in earlier commits, was it accidentally removed?

https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html, instructions under `Bidder Info`

**Comment by asifrahiman** [comment]
> @asifrahiman You need to add a revx.yaml file under static/bidder-info. I see it was present in earlier commits, was it accidentally removed?
> 
> https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html, instructions under `Bidder Info`

@ccorbo It seems to have been an issue with IDE. We have re added the file. Please assist with the review.

**Review by ccorbo — DISMISSED** [review]

File: static/bidder-info/revx.yaml [hunk 23]
```diff
@@ -0,0 +1,17 @@
+endpoint:  "https://prebid-use.atomex.net"
+
+maintainer:
+  email: prebidsupport@revx.io
+
+gvlVendorID: 375
```
Comment (ccorbo) [inline static/bidder-info/revx.yaml:4]: verified
image

**Comment by asifrahiman** [comment]
@bsardo 
Could you review this PR as per your convenience.
Thank you

File: openrtb_ext/bidders.go [hunk 24]
```diff
@@ -261,6 +261,7 @@ var coreBidderNames []BidderName = []BidderName{
 	BidderZeroClickFraud,
 	BidderZetaGlobalSsp,
 	BidderZmaticoo,
+	BidderRevX,
```
Comment (bsardo) [inline openrtb_ext/bidders.go]: Nitpick: please move this up in the list so the list is alphabetical.

File: static/bidder-info/revx.yaml [hunk 25]
```diff
@@ -0,0 +1,17 @@
+endpoint:  "https://prebid-use.atomex.net"
+
+maintainer:
+  email: prebidsupport@revx.io
+
```
Comment (bsardo) [inline static/bidder-info/revx.yaml]: Nitpick: delete blank line

File: static/bidder-info/revx.yaml [hunk 26]
```diff
@@ -0,0 +1,17 @@
+endpoint:  "https://prebid-use.atomex.net"
+
```
Comment (bsardo) [inline static/bidder-info/revx.yaml]: Nitpick: delete blank line

File: openrtb_ext/imp_revx.go [hunk 27]
```diff
@@ -0,0 +1,5 @@
+package openrtb_ext
+
+type ExtImpRevX struct {
+	PubName string `json:"pubName"` // Publisher Name (used for dynamic URL building)
```
Comment (bsardo) [inline openrtb_ext/imp_revx.go]: Is this JSON annotation supposed to be `pubName` or `pubname`?

File: adapters/revx/params_test.go [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+
+	for _, invalidParam := range invalidParams {
+		if err := validator.Validate(openrtb_ext.BidderRevX, json.RawMessage(invalidParam)); err == nil {
+			t.Errorf("Schema accepted invalid RevX params: %s", invalidParam)
+		}
+	}
+}
+
+var validParams = []string{
+	`{"pubname": "publisher123"}`,
```
Comment (bsardo) [inline adapters/revx/params_test.go]: Please add the following test case to test the min length of 1:
`{"pubname": "p"}`,

File: adapters/revx/revx.go [hunk 29]
```diff
@@ -0,0 +1,121 @@
...
+	// Handle specific status codes first
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Received 400 Bad Request from endpoint: %s", a.endPoint),
+		}}
+	}
```
Comment (bsardo) [inline adapters/revx/revx.go]: You can delete this. The call on line 71 to `CheckResponseStatusCodeForErrors` checks for a 400 status code.

File: adapters/revx/revx.go [hunk 30]
```diff
@@ -0,0 +1,121 @@
...
+// MakeBids handles the OpenRTB bid response.
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if externalRequest == nil {
+		return nil, nil
+	}
+
+	// Handle specific status codes first
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
```
Comment (bsardo) [inline adapters/revx/revx.go:62]: Replace this with the adapter helper function:
```
if adapters.IsResponseStatusCodeNoContent(responseData) {
	return nil, nil
}
```

File: adapters/revx/revx.go [hunk 31]
```diff
@@ -0,0 +1,121 @@
...
+
+	if len(typedBids) == 0 {
+		return nil, errs
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(typedBids))
+	bidResponse.Bids = typedBids
+	bidResponse.Currency = serverBidResponse.Cur
+
+	return bidResponse, nil
```
Comment (bsardo) [inline adapters/revx/revx.go]: You should always return `errs` even if there is a typed bid: `return bidResponse, errs`

File: static/bidder-info/revx.yaml [hunk 32]
```diff
@@ -0,0 +1,17 @@
+endpoint:  "https://prebid-use.atomex.net"
```
Comment (bsardo) [inline static/bidder-info/revx.yaml]: This endpoint doesn't appear to be live:
```
curl -i --location --request POST https://prebid-use.atomex.net
HTTP/1.1 504 Gateway Time-out
Date: Fri, 09 Jan 2026 22:36:01 GMT
Content-Type: text/html
Content-Length: 173
Connection: keep-alive

<html>
<head><title>504 Gateway Time-out</title></head>
<body>
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>openresty/1.17.8.2</center>
</body>
</html>
```

**Comment by bsardo** [comment]
@nikithacode please see my latest comments. This PR is close.

**Comment by bsardo** [comment]
Also, please open a docs PR at [prebid.github.io](https://github.com/prebid/prebid.github.io) and provide the link to it in this PR description above.

**Comment by bsardo** [comment]
@asifarahiman please see above.

**Comment by asifrahiman** [comment]
@bsardo 
Please find the docs PR : https://github.com/prebid/prebid.github.io/pull/6403
Please review as per your convenience.

@ccorbo
We have removed the mandatory check for pubname. Please review at your convenience.

Thank you.

File: static/bidder-params/revx.json [hunk 33]
```diff
@@ -0,0 +1,7 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "RevX Adapter Params",
+  "description": "A schema which validates params accepted by the RevX adapter",
+  "type": "object",
+  "properties": {}
```
Comment (ccorbo) [inline static/bidder-params/revx.json:6]: pubname should be a param defined here, and also add param tests. I see you removed it but I'm not sure why

File: static/bidder-info/revx.yaml (hunk 32 above)
Comment (ccorbo) [reply static/bidder-info/revx.yaml]: I'm getting the same thing, it looks like POST requests aren't working but GET requests are
image

File: static/bidder-params/revx.json (hunk 33 above)
Comment (asifrahiman) [reply static/bidder-params/revx.json:6]: [@ccorbo](https://github.com/ccorbo)
We do not require it as a mandatory param anymore. We can proceed without
it.

File: static/bidder-info/revx.yaml (hunk 32 above)
Comment (asifrahiman) [reply static/bidder-info/revx.yaml]: Screenshot 2026-01-27 9 57 17 AM

```
asifabdulrahiman@AFL-IN-LAP577 ~ % curl -i --location --request POST 'https://prebid-use.atomex.net?ag=PREBID'
HTTP/1.1 501 Not Implemented
Date: Tue, 27 Jan 2026 04:29:22 GMT
Content-Type: text/plain; charset=UTF-8
Content-Length: 0
Connection: keep-alive

asifabdulrahiman@AFL-IN-LAP577 ~ % 
```

Hi @bsardo @ccorbo,

The endpoint is live and requires the ag parameter to function correctly. This has been updated in revx.yaml.
At present, the URL will return a 501 response until it is fully enabled on our backend, which is expected until we receive actual supply. Please note that the 501 response is being returned by our backend servers.

If live testing of the flow is required, we can set up an environment to support that.

**Comment by ccorbo** [comment]
@asifarahiman  please review and provide changes for @bsardo 's outstanding comments

File: adapters/revx/revx.go (hunk 29 above)
Comment (asifrahiman) [reply adapters/revx/revx.go]: Done

File: adapters/revx/params_test.go (hunk 28 above)
Comment (asifrahiman) [reply adapters/revx/params_test.go]: Not required as we removed this param

File: openrtb_ext/imp_revx.go (hunk 27 above)
Comment (asifrahiman) [reply openrtb_ext/imp_revx.go]: Not required as we removed this object

File: static/bidder-info/revx.yaml (hunk 26 above)
Comment (asifrahiman) [reply static/bidder-info/revx.yaml]: Done

File: static/bidder-info/revx.yaml (hunk 25 above)
Comment (asifrahiman) [reply static/bidder-info/revx.yaml]: Done

File: openrtb_ext/bidders.go (hunk 24 above)
Comment (asifrahiman) [reply openrtb_ext/bidders.go]: Done

File: adapters/revx/revx.go (hunk 30 above)
Comment (asifrahiman) [reply adapters/revx/revx.go:62]: Done

File: adapters/revx/revx.go (hunk 31 above)
Comment (asifrahiman) [reply adapters/revx/revx.go]: Done

**Comment by asifrahiman** [comment]
@ccorbo 
Updated as per the review comments given by @bsardo

**Review by ccorbo — DISMISSED** [review]

**Comment by bsardo** [comment]
@nikithacode can you please resolve conflicts and update all of the imported package references from `v3` to `v4`?

**Comment by bsardo** [comment]
@nikithacode please resolve conflicts and update the package import versions to `v4`.

**Comment by asifrahiman** [comment]
> @nikithacode can you please resolve conflicts and update all of the imported package references from `v3` to `v4`?

@bsardo Done

**Review by ccorbo — APPROVED** [review]
